packages/@vue/cli/lib/Creator.js

```diff
@@ -238,6 +239,8 @@ module.exports = class Creator {
         error(`Failed fetching remote preset ${chalk.cyan(name)}:`)
         throw e
       }
+    } else if (name.endsWith('.json')) {
+      preset = await readLocalPreset(this.context, name)
```

Comment:
Can directly use `fs-extra` here so it's just `JSON.parse(await fs.readFile())` here, I don't think it's necessary to split into a separate file.

Reply:
It could even just be `preset = await fs.readJson(path.join(this.context, name))`